client/src/components/Connections/ListConnections.tsx

```diff
@@ -4,18 +4,47 @@ import { ConnectionInterface } from '../../App'
 const ListConnections = () => {
   const [connections, setConnections] = useState<any | any[]>([]);
 
-  const getConnections = async () => {
+  // const getConnections = async () => {
+  //   try {
+  //     const response = await fetch("http://localhost:3000/connections");
+  //     const jsonData = await response.json();
+
+  //     const tuples: any[] = [];
+  //     await jsonData.rows.map(async (connection:ConnectionInterface) => {
+  //       const u1 = await getUserNamePromise(connection.user1_id);
+  //       const u2 = await getUserNamePromise(connection.user2_id);
+  //       tuples.push({ user1_id: connection.user1_id, user1_name: u1, user2_name: u2, user2_id: connection.user2_id });
+  //       setConnections(tuples);
+  //     });
+
+  //   } catch (err) {
+  //     if(err instanceof Error){
+  //       console.error(err.message);
+  //   }else{
+  //       console.error("Unexpected error",err);
+  //   }
+  //   }
+  // };
+
+  const getConnections =  () => {
     try {
-      const response = await fetch("http://localhost:3000/connections");
-      const jsonData = await response.json();
+      const response = fetch("http://localhost:3000/connections");
+      response.then((res) => {
+        const jsonData =  res.json();
+        jsonData.then((resJson) => {
+          const tuples: any[] = [];
+          resJson.rows.map( (connection:ConnectionInterface) => {
+            const name1Promise =  getUserNamePromise(connection.user1_id);
+            const name2Promise =  getUserNamePromise(connection.user2_id);
 
-      const tuples: any[] = [];
-      await jsonData.rows.map(async (connection:ConnectionInterface) => {
-        const u1 = await getUserNamePromise(connection.user1_id);
-        const u2 = await getUserNamePromise(connection.user2_id);
-        tuples.push({ user1_id: connection.user1_id, user1_name: u1, user2_name: u2, user2_id: connection.user2_id });
-        setConnections(tuples);
-      });
+            Promise.all([name1Promise,name2Promise]).then((values) => {
+              console.log(values);
+              tuples.push({ user1_id: connection.user1_id, user1_name: values[0], user2_name: values[1], user2_id: connection.user2_id });
+              if(tuples.length == resJson.rows.length) setConnections(tuples); 
+            });          
```

Comment:
I'm not sure this will solve the syncing issues you were facing previously where sometimes all the connections were updated and sometimes only a few of them. Did you check it?

I think it will be better to use the returned array of the map to generate an array with all the promises and wait for them outside. Wdyt?